From d5bc03738fd99fff9e9d98817c05b1d802ea0edb Mon Sep 17 00:00:00 2001
From: Ping Sun <ping.x.sun@sony.com>
Date: Fri, 25 May 2018 15:00:48 +0800
Subject: [PATCH 3/8] Fix an issue wrong network operator name is displayed on
 MSIM devices

SIM1 operator network name is displayed as SIM2 operator network name on
Mobile Network setting because TelephonyManager#getNetworkOperatorName
with no subId is always used to set the summary text.

To resolve this issue, use TelephonyManager#getNetworkOperatorName with
subId to get operator network name based on SIM.

Bug: 109936826
Test: manual - Checked that the network name is displayed according to
the SIM network operator
Change-Id: Ief6d5148ef0e27bb31d869994cf64ba26ce4f7d4
---
 src/com/android/phone/NetworkSelectListPreference.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/com/android/phone/NetworkSelectListPreference.java b/src/com/android/phone/NetworkSelectListPreference.java
index 341ac8d2b..3d932b0b5 100644
--- a/src/com/android/phone/NetworkSelectListPreference.java
+++ b/src/com/android/phone/NetworkSelectListPreference.java
@@ -325,7 +325,7 @@ public class NetworkSelectListPreference extends ListPreference
         TelephonyManager telephonyManager = (TelephonyManager)
                 getContext().getSystemService(Context.TELEPHONY_SERVICE);
 
-        setSummary(telephonyManager.getNetworkOperatorName());
+        setSummary(telephonyManager.getNetworkOperatorName(mSubId));
 
         setOnPreferenceChangeListener(this);
     }
-- 
2.17.1

